cherrypy/wsgiserver/__init__.py

```diff
@@ -1695,7 +1695,7 @@ def stop(self, timeout=5):
 
         # Don't join currentThread (when stop is called inside a request).
         current = threading.currentThread()
-        if timeout and timeout >= 0:
+        if timeout is not None and timeout >= 0:
```

Comment:
I don't know. It's needed to check tests. Make your suggestion. You may even need to add some new test suite.